cmd/maya-apiserver/spc-watcher/lease.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2017 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package spc
+
+import (
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"strings"
+	clientset "github.com/openebs/maya/pkg/client/generated/clientset/internalclientset"
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/runtime"
+)
+
+const (
+	// SpcLeaseKey is the key that will we used to acquire lease on spc object.
+	// It will be present in spc annotations
+	// If key has an empty value, that means no one has acquired a lease on spc object.
+	SpcLeaseKey = "openebs.io/spc-lease"
+	// SpcLeaseValue is the value of the SpcLeaseKey
+	SpcLeaseValue = "acquired"
+)
+
+// Leases is an interface which assists in getting and releasing lease over an spc object
+type Leases interface {
+	// GetLease will try to get a lease on spc, in case of failure it will return error
+	GetLease() (string, error)
+	// UpdateLease will update the lease value of the spc
+	UpdateLease() (string, error)
```

Comment:
I guess, interface methods & struct methods do not match.
However, will it be possible/better to have different interface or higher order function for each CRUD operation.

Reply:
I will fix it.

Reply:
About higher order functions . We can discuss.
If it gives us benefit for our use case I will implement it.

Reply:
Fixed. But about 3 separate interfaces,we will see it in upcoming PR after more discussions.